### PR TITLE
Kad. Routing Fixes

### PR DESCRIPTION
**Pull request by ejfitzgerald**
- Fix issues with kad. routing selection
- Correct issue with kad routing "black hole" (merge issue)
- Introduce parameter to `/api/status/muddle?network=IHUB` to add filtering of muddle instances
- Fix busy wait state machine issue with DAG sync

### CODE REVIEW

File: libs/ledger/src/dag/dag_sync_service.cpp [hunk 1]
```diff
@@ -189,6 +189,7 @@ DAGSyncService::State DAGSyncService::OnResolveMissing()
     }
   }
 
+  state_machine_->Delay(std::chrono::milliseconds{100});
```
Comment (ejfitzgerald) [inline libs/ledger/src/dag/dag_sync_service.cpp:192]: Unrelated in the main but fixes 100% CPU busy state machine issue

File: libs/muddle/src/router.cpp [hunk 2]
```diff
@@ -558,7 +558,13 @@ Router::UpdateStatus Router::AssociateHandleWithAddress(Handle
...
   assert(handle);
 
-  bool display{false};
+  if (broadcast)
+  {
+    FETCH_LOG_TRACE(logging_name_, "Ignoring routing messages from broadcast");
+    return status;
+  }
+
+  bool display{true};
```
Comment (ejfitzgerald) [inline libs/muddle/src/router.cpp]: Whoops!

File: libs/muddle/src/muddle_status.cpp [hunk 3]
```diff
@@ -125,12 +125,42 @@ void BuildMuddleStatus(Muddle const &muddle, variant::Variant &output)
...
 
-}  // namespace
+MuddleRegistry::MuddleMap FilterInstances(MuddleRegistry::MuddleMap const &map,
+                                          std::string const &              target_network)
+{
+  MuddleRegistry::MuddleMap filtered_map{};
+
+  for (auto const &instance : map)
+  {
+    auto const network = instance.second.lock();
```
Comment (n-hutton) [inline libs/muddle/src/muddle_status.cpp:178]: What is going on here with this lock?

File: libs/muddle/src/muddle_status.cpp [hunk 4]
```diff
@@ -125,12 +125,42 @@ void BuildMuddleStatus(Muddle const &muddle, variant::Variant &output)
   BuildPeerSelection(muddle.peer_selector(), output["peerSelection"]);
 }
 
-}  // namespace
+MuddleRegistry::MuddleMap FilterInstances(MuddleRegistry::MuddleMap const &map,
```
Comment (n-hutton) [inline libs/muddle/src/muddle_status.cpp:171]: Might be nice for a few dox comments

File: libs/muddle/src/peer_selector.cpp [hunk 5]
```diff
@@ -397,6 +411,16 @@ void PeerSelector::MakeAnnouncement()
 {
   if (announcement_interval_.HasExpired())
   {
+    if (endpoint_.GetDirectlyConnectedPeerSet().empty())
```
Comment (n-hutton) [inline libs/muddle/src/peer_selector.cpp:414]: Basically when there are no peers, try harder? If so, awesome!

File: libs/muddle/src/router.cpp [hunk 6]
```diff
@@ -825,16 +820,16 @@ void Router::RoutePacket(PacketPtr const &packet, bool external)
...
-    }
-
+    // if kad routing is enabled we should use this to route packets
     if (kademlia_routing_)
     {
       handle = LookupKademliaClosestHandle(packet->GetTarget());
-      return;
+      if (handle != 0u)
+      {
+        FETCH_LOG_WARN(logging_name_, "kad routing packet");
```
Comment (n-hutton) [inline libs/muddle/src/router.cpp]: is this a warn? Also, logging_name_ ?

**Review by n-hutton — CHANGES_REQUESTED** [review]

File: libs/muddle/src/muddle_status.cpp (hunk 3 above)
Comment (ejfitzgerald) [reply libs/muddle/src/muddle_status.cpp:178]: It is a weak_ptr

File: libs/muddle/src/router.cpp (hunk 6 above)
Comment (ejfitzgerald) [reply libs/muddle/src/router.cpp]: Nope good spot

**Review by n-hutton — APPROVED** [review]

**Pull request merged**